runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -1186,24 +1195,25 @@ JITServerAOTCache::writeCache(FILE *f) const
       }
 
    if (1 != fwrite(&header, sizeof(JITServerAOTCacheHeader), 1, f))
-      return false;
+      return 0;
 
    if (!writeRecordList(f, _classLoaderHead, header._numClassLoaderRecords))
-      return false;
+      return 0;
    if (!writeRecordList(f, _classHead, header._numClassRecords))
-      return false;
+      return 0;
    if (!writeRecordList(f, _methodHead, header._numMethodRecords))
-      return false;
+      return 0;
    if (!writeRecordList(f, _classChainHead, header._numClassChainRecords))
-      return false;
+      return 0;
    if (!writeRecordList(f, _wellKnownClassesHead, header._numWellKnownClassesRecords))
-      return false;
+      return 0;
    if (!writeRecordList(f, _aotHeaderHead, header._numAOTHeaderRecords))
-      return false;
+      return 0;
    if (!writeCachedMethodList(f, _cachedMethodHead, header._numCachedAOTMethods))
-      return false;
+      return 0;
 
-   return true;
+   TR_ASSERT_FATAL(header._numCachedAOTMethods != 0, "Expected to write at least one method to the AOT cache file");
```

Comment:
Nitpick: this assertion should be done before we write out the header.

Reply:
Done